app/ts/McdaApp/Workspace/CurrentTab/ResultsTabs/DeterministicTab/DeterministicResultsContext/DeterministicResultsContext.tsx

```diff
@@ -41,57 +32,29 @@ export function DeterministicResultsContextProviderComponent({
 }: {
   children: any;
 }): JSX.Element {
-  const {setError} = useContext(ErrorContext);
-  const {scales} = useContext(WorkspaceContext);
-  const {
-    filteredCriteria,
-    filteredAlternatives,
-    filteredEffects,
-    filteredWorkspace
-  } = useContext(CurrentSubproblemContext);
   const {pvfs, currentScenario} = useContext(CurrentScenarioContext);
-
-  const [sensitivityTableValues, setSensitivityTableValues] = useState<
-    Record<string, Record<string, ISensitivityValue>>
-  >(
-    getInitialSensitivityValues(
-      filteredCriteria,
-      filteredAlternatives,
-      filteredEffects,
-      scales
-    )
+  const {setError} = useContext(ErrorContext);
+  const {filteredAlternatives, filteredWorkspace} = useContext(
+    CurrentSubproblemContext
   );
+
   const [recalculatedCells, setRecalculatedCells] = useState<
     IRecalculatedCell[]
   >([]);
-  const [weights, setWeights] = useState<IWeights>();
-  const [baseTotalValues, setBaseTotalValues] =
-    useState<Record<string, number>>();
-  const [baseValueProfiles, setBaseValueProfiles] =
-    useState<Record<string, Record<string, number>>>();
   const [recalculatedTotalValues, setRecalculatedTotalValues] =
     useState<Record<string, number>>();
   const [recalculatedValueProfiles, setRecalculatedValueProfiles] =
     useState<Record<string, Record<string, number>>>();
+  const [recalculatedWeights, setRecalculatedWeights] = useState<
```

Comment:
Name is confusing, I feel it currently suggest this value is set after clicking recalculate, while it is being used to do the recalculation of the value profiles